From 03369c3e59e803d55fdf52da16aaffb0da53985e Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 13:12:44 -0400
Subject: [ethernet] i40e/i40evf: add ASQ write back timeout variable to AQ structure

Message-id: <1409663591-32531-122-git-send-email-sassmann@redhat.com>
Patchwork-id: 90594
O-Subject: [RHEL7.1 PATCH 121/148] i40e/i40evf: add ASQ write back timeout variable to AQ structure
Bugzilla: 1091127
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Kamil Krawczyk <kamil.krawczyk@intel.com>

Add new variable defining ASQ command write back timeout to allow for
dynamic modification of this timeout. Initialize it on AQ initialize
routine with default value, vary it on device ID.

Change-ID: I5c9908f9d7c5455634353b694a986d6f146d1b9d
Signed-off-by: Kamil Krawczyk <kamil.krawczyk@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 09c4e56b3cddecb6cac70339dcec63ed1b4e18c2)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40evf/i40e_adminq.c b/drivers/net/ethernet/intel/i40evf/i40e_adminq.c
index c277763..cc4b6db 100644
--- a/drivers/net/ethernet/intel/i40evf/i40e_adminq.c
+++ b/drivers/net/ethernet/intel/i40evf/i40e_adminq.c
@@ -567,6 +567,9 @@ i40e_status i40evf_init_adminq(struct i40e_hw *hw)
  /* Set up register offsets */
  i40e_adminq_init_regs(hw);
 
+ /* setup ASQ command write back timeout */
+ hw->aq.asq_cmd_timeout = I40E_ASQ_CMD_TIMEOUT;
+
  /* allocate the ASQ */
  ret_code = i40e_init_asq(hw);
  if (ret_code)
@@ -814,7 +817,7 @@ i40e_status i40evf_asq_send_command(struct i40e_hw *hw,
    /* ugh! delay while spin_lock */
    udelay(delay_len);
    total_delay += delay_len;
-  } while (total_delay <  I40E_ASQ_CMD_TIMEOUT);
+  } while (total_delay <  hw->aq.asq_cmd_timeout);
  }
 
  /* if ready, copy the desc back to temp */
diff --git a/drivers/net/ethernet/intel/i40evf/i40e_adminq.h b/drivers/net/ethernet/intel/i40evf/i40e_adminq.h
index 9335375..1628455 100644
--- a/drivers/net/ethernet/intel/i40evf/i40e_adminq.h
+++ b/drivers/net/ethernet/intel/i40evf/i40e_adminq.h
@@ -84,6 +84,7 @@ struct i40e_arq_event_info {
 struct i40e_adminq_info {
  struct i40e_adminq_ring arq;    /* receive queue */
  struct i40e_adminq_ring asq;    /* send queue */
+ u32 asq_cmd_timeout;            /* send queue cmd write back timeout*/
  u16 num_arq_entries;            /* receive queue depth */
  u16 num_asq_entries;            /* send queue depth */
  u16 arq_buf_size;               /* receive queue buffer size */
-- 
1.7.1